agent/connectionchecker/connection_checker.go

```diff
@@ -127,7 +128,12 @@ func (cc *ConnectionChecker) checkMySQLConnection(ctx context.Context, dsn strin
 	defer db.Close() //nolint:errcheck
 
 	if err = cc.sqlPing(ctx, db); err != nil {
-		res.Error = err.Error()
+		switch err.(type) {
+		case x509.HostnameError:
+			res.Error = fmt.Sprintf("mysql ssl certificate is misconfigured: %s. Make sure the certificate includes the requested hostname/IP in CN or subjectAltName fields", err.Error())
```

Comment:
It already includes the original message. I could remove the `mysql ssl certificate is misconfigured`, then the message would be something like `Connection check failed: x509: cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs. Make sure the certificate includes the requested hostname/IP in CN or subjectAltName fields` but I'm not sure if it would still be easy to understand for the user

Reply:
I think we souldn't invent something new, just use `errors.Wrap` function.